std/assembly/math.ts

```diff
@@ -476,16 +474,17 @@ export namespace NativeMath {
     u = (0x3FF + k) << 52;
     var twopk = reinterpret<f64>(u);
     var y: f64;
-    if (k < 0 || k > 56) {
+    if (<i32>(k < 0) | <i32>(k > 56)) {
```

Comment:
I *think* we should make the compiler smarter about logical ors instead of convoluting the source like this. Wdyt?

Reply:
Definitely yes! I even have opened proposal for that:  https://github.com/AssemblyScript/assemblyscript/issues/277

Reply:
So, can we keep the `||` here and at the other places, if any, for now?

Reply:
sure